executor/importer/chunk_process.go

```diff
@@ -372,11 +414,7 @@ func (w *IndexRouteWriter) AppendRows(ctx context.Context, _ []string, rows enco
 		if err != nil {
 			return errors.Trace(err)
 		}
-		writer, ok := w.writers[indexID]
-		if !ok {
-			writer = w.writerFactory(indexID)
-			w.writers[indexID] = writer
-		}
+		writer := w.getWriter(indexID)
 		if err = writer.WriteRow(ctx, item.Key, item.Val, nil); err != nil {
```

Comment:
seems if one writer has entered `WriteRow`, it will block other `kvs` which also need to access this writer. We can improve it later.

Reply:
do you mean `kvs` in this loop? or in other concurrency?

Reply:
other call of WriteRow

Reply:
depends on whether it's same index, other concurrency can still use writers of other index